rest/doc_api.go

```diff
@@ -370,14 +370,25 @@ func (h *handler) handlePutDocReplicator2(docid string, roundTrip bool) (err err
 	if err != nil {
 		return err
 	}
-	if bodyBytes == nil || len(bodyBytes) == 0 {
-		return base.ErrEmptyDocument
-	}
 
-	newDoc := &db.Document{
-		ID: docid,
+	hasAttachments := bytes.Contains(bodyBytes, []byte(db.BodyAttachments))
+	hasExpiry := bytes.Contains(bodyBytes, []byte(db.BodyExpiry))
+
+	var body db.Body
+	var newDoc *db.IncomingDocument
+	newDoc = db.NewIncomingDocument(bodyBytes)
+
+	if hasAttachments || hasExpiry {
+		// In handleRev the below is actually using ReadJSONBody
+		err := base.JSONUnmarshal(bodyBytes, &body)
+		if err != nil {
+			return base.HTTPErrorf(http.StatusBadRequest, "error occurred reading JSON Body: %v", err)
+		}
+		newDoc, err = body.ToIncomingDocument()
+		if err != nil {
+			return base.HTTPErrorf(http.StatusBadRequest, "error occurred building IncomingDocument from body: %v", err)
```

Comment:
Same - This doesn't seem like a client-side error.

Reply:
Will go back to this once handleRev is done....